File: src/v3xctrl_ui/utils/Commands.py
from v3xctrl_control.message import Command


class Commands:
    @staticmethod
    def service(name: str, action: str) -> Command:
        return Command("service", {"name": name, "action": action})

    @staticmethod
    def video_start() -> Command:
        return Commands.service("v3xctrl-video", "start")

    @staticmethod
    def video_stop() -> Command:
        return Commands.service("v3xctrl-video", "stop")

    @staticmethod
    def shell_start() -> Command:
        return Commands.service("v3xctrl-reverse-shell", "start")

    @staticmethod
    def shell_stop() -> Command:
        return Commands.service("v3xctrl-reverse-shell", "stop")

    @staticmethod
    def recording_start() -> Command:
        return Command("recording", {"action": "start"})

    @staticmethod
    def recording_stop() -> Command:
        return Command("recording", {"action": "stop"})

    @staticmethod
    def trim_increase() -> Command:
        return Command("trim", {"action": "increase"})

    @staticmethod
    def trim_decrease() -> Command:
        return Command("trim", {"action": "decrease"})

    @staticmethod
    def shutdown() -> Command:
        return Command("shutdown")

    @staticmethod
    def restart() -> Command:
        return Command("restart")

File: android/app/src/main/java/com/v3xctrl/viewer/messages/Commands.kt
package com.v3xctrl.viewer.messages

/**
 * Factory object for creating common Command messages.
 */
object Commands {
    fun videoStart() = Command(command = "service", parameters = mapOf("action" to "start", "name" to "v3xctrl-video"))
    fun videoStop() = Command(command = "service", parameters = mapOf("action" to "stop", "name" to "v3xctrl-video"))
    fun recordingStart() = Command(command = "recording", parameters = mapOf("action" to "start"))
    fun recordingStop() = Command(command = "recording", parameters = mapOf("action" to "stop"))
    fun trimIncrease() = Command(command = "trim", parameters = mapOf("action" to "increase"))
    fun trimDecrease() = Command(command = "trim", parameters = mapOf("action" to "decrease"))
    fun shutdown() = Command(command = "shutdown")
    fun restart() = Command(command = "restart")
}
